### PR TITLE
created api for user group roles id and added integration test

### PR DESCRIPTION
**Pull request by meanwhile7**
### Closes: 
- https://github.com/Real-Dev-Squad/website-dashboard/issues/479

## Test coverage
Screenshot from 2023-09-08 21-54-26

### Description
Api returns the group roleId of groups user is part of

### CODE REVIEW

File: models/discordactions.js [hunk 1]
```diff
@@ -74,6 +74,26 @@ const isGroupRoleExists = async (rolename) => {
...
+    const userRolesSnapshot = await memberRoleModel.where("userid", "==", userId).get();
+
+    const userRoles = userRolesSnapshot.docs.map((doc) => ({
+      ...doc.data(),
+    }));
+    const userRolesObject = {
+      userId: userId,
+      groups: userRoles.map((userRole) => ({
+        roleId: userRole.roleid,
+      })),
```
Comment (pallabez) [inline models/discordactions.js:88]: I think we should add all the group data here. What do you think?

File: routes/discordactions.js [hunk 2]
```diff
@@ -17,6 +18,7 @@ const router = express.Router();
 router.post("/groups", authenticate, checkIsVerifiedDiscord, validateGroupRoleBody, createGroupRole);
 router.get("/groups", authenticate, checkIsVerifiedDiscord, getAllGroupRoles);
 router.post("/roles", authenticate, checkIsVerifiedDiscord, validateMemberRoleBody, addGroupRoleToMember);
+router.get("/user/group/roles", authenticate, checkIsVerifiedDiscord, getGroupsRoleId);
```
Comment (shubhamsigdar1) [inline routes/discordactions.js:21]: please discuss with others and change route

**Review by pallabez — APPROVED** [review]
LGTM

File: controllers/discordactions.js [hunk 3]
```diff
@@ -86,6 +85,19 @@ const getAllGroupRoles = async (req, res) => {
   }
 };
 
+const getGroupsRoleId = async (req, res) => {
+  try {
+    const { discordId } = req.userData;
+    const userRolesArray = await discordRolesModel.getGroupRolesForUser(discordId);
```
Comment (prerana1821) [inline controllers/discordactions.js:91]: I think userGroupRoles or just userRoles would be a good name. Having array in the name is not required.

**Review by prerana1821 — CHANGES_REQUESTED** [review]
One small comment. Looks good otherwise.

File: models/discordactions.js [hunk 4]
```diff
@@ -74,6 +74,26 @@ const isGroupRoleExists = async (rolename) => {
   }
 };
 
+const getGroupRolesForUser = async (userId) => {
+  try {
+    const userRolesSnapshot = await memberRoleModel.where("userid", "==", userId).get();
```
Comment (pallabez) [inline models/discordactions.js:79]:
```suggestion
const getGroupRolesForUser = async (discordId) => {
  try {
    const userRolesSnapshot = await memberRoleModel.where("userid", "==", discordId).get();
```

**Review by shubhamsigdar1 — COMMENTED** [review]
Add test coverage on description

**Review by pallabez — DISMISSED** [review]
LGTM

File: controllers/discordactions.js (hunk 3 above)
Comment (meanwhile7) [reply controllers/discordactions.js]: Done changes has been done

File: routes/discordactions.js (hunk 2 above)
Comment (meanwhile7) [reply routes/discordactions.js:21]: Shubham test coverage you  mean adding details regarding the test in the description ?

File: models/discordactions.js
Comment (vinit717) [inline models/discordactions.js:1]: Please write the unit test for this.

**Review by vinit717 — CHANGES_REQUESTED** [review]

**Review by prerana1821 — DISMISSED** [review]

File: routes/discordactions.js (hunk 2 above)
Comment (meanwhile7) [reply routes/discordactions.js]: Done

**Review by gauravsinhaweb — APPROVED** [review]
LGTM

**Review by prerana1821 — APPROVED** [review]

File: models/discordactions.js
Comment (meanwhile7) [reply models/discordactions.js:1]: Its to be included in next pr

**Pull request merged**